### PR TITLE
feat: Adjust checkout for digital products

### PR DESCRIPTION
**Pull request by soniaklimas**
I want to merge this change because it adjusts checkout for digital products - if there are only digital products in the cart, checkout page excludes unnecessary steps.

# Pull Request Checklist

- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

File: packages/infrastructure/src/checkout/saleor/infrastructure/checkout-get-infra.ts [hunk 1]
```diff
@@ -50,6 +50,7 @@ const serializeCheckout = (checkout: CheckoutFragment): Checkout => {
 
   return {
     ...checkout,
+    isShippingRequired: !!checkout.isShippingRequired,
```
Comment (GrzegorzDerdak) [inline packages/infrastructure/src/checkout/saleor/infrastructure/checkout-get-infra.ts:53]: Why the `!!` and not just `isShippingRequired: checkout.isShippingRequired`?

**Review by GrzegorzDerdak — APPROVED** [review]
